### ISSUE TITLE
Export FILENAME as environment variable

### ISSUE THREAD

**Issue by thomasfl** [kimmobrunfeldt/chokidar-cli#40]
I suggest making the updated FILENAME available as an environment variable from within the command. It makes is possible to run commands like this:

```
  $ chokidar "**/**.js" -c "jshint $FILENAME"
```

The very much similar filewatcher CLI, written in ruby, exports these environment variables:

```
FILENAME           Relative filename.
ABSOLUTE_FILENAME  Asolute filename.
RELATIVE_FILENAME  Same as FILENAME but starts with "./"
BASENAME           File basename.
EVENT              Event type. Is either 'changed', 'delete' or 'new'.
DIRNAME            Absolute directory name.
```

Source: https://github.com/thomasfl/filewatcher

BTW! Chokidar seems to be a very good piece of software.

**Comment by es128** [comment]
From the readme:

#### Pass the path and event details in to your custom command

```
chokidar '**/*.less' -c 'if [ "{event}" = "change" ]; then npm run build-less -- {path}; fi;'
```

**Comment by thomasfl** [comment]
Thank's @es128. I didn't notice that feature in the readme.

I would however suggest that in instead of replacing "{path}" in the command string like this:

```
opts.command
                    .replace(/\{path\}/ig, path)
                    .replace(/\{event\}/ig, event)
```

...that path was made available as an environment variable for the command by passing the environment variables to spawn like this:

```
child = childProcess.spawn(SHELL_PATH, [EXECUTE_OPTION, cmd], {
                cwd: opts.cwd,
                stdio: opts.pipe ? 'inherit' : null,
                env: {PATH: path, EVENT: event}
            });
```

If there's any interest in this feature, I'll be happy to create a pull request.

**Comment by kimmobrunfeldt** [comment]
@thomasfl I agree, using a random string template format is worse than using environment variables.

I'll leave this open. I'm open to merge a PR which implements this.

**Comment by es128** [comment]
If you're going to replace the original templating with this, be mindful to release with a major version bump.

**Comment by kimmobrunfeldt** [comment]
@es128 yeah good point. Will do

**Comment by thomasfl** [comment]
This was way more complicated than expected to implement. The spawn function is wrapped in four layers of lodash throttle magic. I was simply not able to figure out how to pass another variable to utils.run() from the watcher-loop. I suggest the environment variables are named FILEPATH and EVENT.
